scandium-core/src/main/java/org/eclipse/californium/scandium/util/ListUtils.java

```diff
@@ -0,0 +1,53 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Bosch Software Innovations GmbH and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ ******************************************************************************/
+package org.eclipse.californium.scandium.util;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Utility class to create a unmodifiable "ordered set" as {@link List}.
+ */
+public class ListUtils {
+
+	/**
+	 * Initialize ordered values, testing for contained item duplicates.
+	 * 
+	 * @param values list of values. Maybe {@code null} for empty list.
+	 * @return unmodifiable list with unique items.
+	 * @throws IllegalArgumentException if duplicate items are contained
+	 */
+	public static <T> List<T> init(List<T> values) {
```

Comment:
I would not keep the `init` method and rather change it by `hasDuplicate()` or `containsDuplicate()`.

But this is clearly a detail, change it only if it makes sense to you.